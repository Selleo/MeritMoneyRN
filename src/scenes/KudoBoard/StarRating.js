import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { TouchableOpacity, View, StyleSheet } from 'react-native'

import Icon from 'src/components/Icon'
import { yellow } from 'src/styles/colors'

export default class StarRating extends Component {
  static propTypes = {
    quantity: PropTypes.number.isRequired,
    setQuantity: PropTypes.func.isRequired,
  }

  _setKudoAmount = kudoAmount => () => this.setState({ kudoAmount })

  renderStarts = () => {
    const { quantity, setQuantity } = this.props
    const stars = []

    for (let i = 1; i < 6; i++) {
      stars.push(
        <TouchableOpacity key={i} onPress={setQuantity(i)}>
          <Icon
            color={yellow}
            name={quantity >= i ? 'star-full2' : 'star-empty3'}
            size={28}
            style={styles.kudo}
          />
        </TouchableOpacity>,
      )
    }
    return stars
  }

  render() {
    return <View style={styles.container}>{this.renderStarts()}</View>
  }
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    marginVertical: 5,
  },
  kudo: {
    paddingHorizontal: 8,
  },
})
