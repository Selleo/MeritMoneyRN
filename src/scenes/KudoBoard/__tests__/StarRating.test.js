import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'

import StarRating from '../StarRating'

describe('<StarRating />', () => {
  const defaultProps = {
    quantity: 1,
    setQuantity: jest.fn(),
  }

  const wrapper = renderer.create(<StarRating {...defaultProps} />)

  test('render', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
