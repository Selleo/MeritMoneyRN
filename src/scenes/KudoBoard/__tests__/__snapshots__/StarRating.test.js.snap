// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StarRating /> render 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "marginVertical": 5,
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#fcda53",
            "fontSize": 28,
          },
          Object {
            "paddingHorizontal": 8,
          },
          Object {
            "fontFamily": "icomoon",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#fcda53",
            "fontSize": 28,
          },
          Object {
            "paddingHorizontal": 8,
          },
          Object {
            "fontFamily": "icomoon",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#fcda53",
            "fontSize": 28,
          },
          Object {
            "paddingHorizontal": 8,
          },
          Object {
            "fontFamily": "icomoon",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#fcda53",
            "fontSize": 28,
          },
          Object {
            "paddingHorizontal": 8,
          },
          Object {
            "fontFamily": "icomoon",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#fcda53",
            "fontSize": 28,
          },
          Object {
            "paddingHorizontal": 8,
          },
          Object {
            "fontFamily": "icomoon",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
